demo_game.py

```diff
@@ -11,58 +12,61 @@
 """)
 
 mallet = Item('rusty mallet', 'mallet')
-valley.items = Bag({mallet,})
-
-inventory = Bag()
+valley.items = Bag({mallet, })
 
 
 @when('north', direction='north')
 @when('south', direction='south')
 @when('east', direction='east')
 @when('west', direction='west')
-def go(direction):
-    global current_room
-    room = current_room.exit(direction)
+def go(direction, game):
```

Comment:
I believe I attempted to make it so that functions without a `game` parameter still work BTW

Reply:
I believe that much of the reasoning for me behind not making `game` a global is that it allows the user to have multiple games running in the same file. While this may seem like an unnecessary feature, I can't tell you how many times I've been tripped up when I discovered that the library I'm using prevents me from having multiple of whatever-it-helps-me-to-implement in one program.

Reply:
I agree that supporting multiple games from the same file would be better for more serious applications, but for education I just don't think it's necessary.

I could imagine making global state optional, by creating classes that represent all state that is currently global, and preserving the same API by having "default" instances of those classes. But really I think the extra complexity of that is not justified by its usefulness for education.

In particular, I did think about approaches for running adventurelib games in the browser, and while a backend webapp might be best to allow multiple games in the same interpreter, my preferred approach would be #11 - get the Python code working client-side in the browser.